tools/prepare_release.csx

```diff
@@ -57,31 +67,85 @@ public static string ReadGitHubToken()
+    var issueRequest = new RepositoryIssueRequest { Milestone = milestone.Number.ToString(), State = ItemStateFilter.All };
+    var issues = (await gitHubClient.Issue.GetAllForRepository(repoOwner, repoName, issueRequest)).ToList();
+    Console.WriteLine($"Fetched {issues.Count} issues in milestone '{milestone.Title}'");
+    return issues;
+}
+
+public static IEnumerable<string> GetIssuesReferencedFromRelease(Release release)
+{
+    var issuesReferencedFromRelease = new HashSet<string>();
+    foreach (Match match in Regex.Matches(release.Body, @"\(#([0-9]+)\)"))
+    {
+        issuesReferencedFromRelease.Add(match.Groups[1].Value);
+    }
+    return issuesReferencedFromRelease;
+}
+
+public static Issue PopExistingReleaseIssue(IList<Issue> issues)
 {
-    Console.WriteLine("Fetching release issue 'Release vNext'...");
-    var issueRequest = new RepositoryIssueRequest { Milestone = existingMilestone.Number.ToString(), State = ItemStateFilter.Open };
-    var existingIssue = (await gitHubClient.Issue.GetAllForRepository(repoOwner, repoName, issueRequest))
-        .Single(issue => issue.Title == "Release vNext");
-    Console.WriteLine($"Fetched release issue #{existingIssue.Number}: '{existingIssue.Title}'");
-    return existingIssue;
+    var issue = issues.Single(i => i.Title == "Release " + existingReleaseName);
+    Console.WriteLine($"Found release issue #{issue.Number}: '{issue.Title}'");
+    issues.Remove(issue);
+    return issue;
+}
+
+public static void CrossReferenceIssues(IEnumerable<Issue> issuesInMilestone, IEnumerable<string> issueNumbersReferencedFromRelease)
+{
+    var issueNumbersInMilestone = issuesInMilestone.Select(i => i.Number.ToString());
+    var issueNumbersInReleaseButNotMilestone = issueNumbersReferencedFromRelease.Except(issueNumbersInMilestone);
+    var issuesInMilestoneButNotRelease = issuesInMilestone.Where(i => !issueNumbersReferencedFromRelease.Contains(i.Number.ToString()));
+
+    if (!issuesInMilestoneButNotRelease.Any() && !issueNumbersInReleaseButNotMilestone.Any())
+    {
+        Console.WriteLine("The release refers to the same issues included in the milestone. Congratulations.");
+        return;
+    }
+
+    Console.WriteLine();
+
+    if (issuesInMilestoneButNotRelease.Any())
+    {
+        Console.WriteLine("The following issues are linked to the milestone but not referenced in the release:");
+        foreach (var issue in issuesInMilestoneButNotRelease)
+        {
+            Console.WriteLine($"  #{issue.Number} - {issue.Title}");
+        }
+
+        Console.WriteLine();
+    }
+
+    if (issueNumbersInReleaseButNotMilestone.Any())
+    {
+        Console.WriteLine("The following issues are referenced in the release but not linked to the milestone:");
+        foreach (var issueNumber in issueNumbersInReleaseButNotMilestone)
+        {
+            Console.WriteLine($"  #{issueNumber}");
+        }
+
+        Console.WriteLine();
+    }

```

Comment:
It's an option, but I was afraid to make it so. The main reason is that we don't make a new milestone for pre-release versions. Suppose we're preparing 1.2.3-beta02, which contains issues 4, 5, and 6. Issues 1, 2, and 3 will have already been released in 1.2.3-beta01. But the milestone will have issues 1, 2, 3, 4, 5, and 6.

Options:
1. make the script smarter so it can deal with pre-releases. I think this would be very hard
2. add a "force" flag or some other way to override the failure
3. let the release happen, but move the "mismatch" message to the end of the output so they're more easily seen
4. let the release happen, but pause for input, forcing the user to acknowledge that there's a mismatch. I've used this technique in [other "prepare release" situations.](https://github.com/blairconrad/SelfInitializingFakes/blob/cfc6d3242248143f5011a8273411b21b4d108378/tools/release-version.ps1#L24-L31) .

Me, I like 3 or 4. I suppose we could also add colour to the output, to make it "pop" (not "take". Ha!), but I don't think it'll help that much.

Reply:
> let the release happen, but pause for input, forcing the user to acknowledge that there's a mismatch. I've used this technique in other "prepare release" situations. .

Sounds good to me 👍